### PR TITLE
Skip load_state key modification in training mode

### PR DESCRIPTION
**Pull request by MarshallYan**
### Associated Issue(s)
 - [x] Key modification in load_state_dict() in training mode causing an error #264.

## Pull Request Checklist
 - [x] Issue(s) raised/addressed and linked
 - [x] PR is ready for review

### CODE REVIEW

**Comment by wiederm** [comment]
@MarshallYan , I am not sure if this modification is necessary.

I reverted the modification that you made to the `load_state_dict` method and outlined the 3 use cases in this [test](https://github.com/choderalab/modelforge/blob/b81639f015f7d912f66338ec801694a923d38119/modelforge/tests/test_models.py#L343). They are currently all passing. Could you please first add the use case to this test and check that it is failing.

**Comment by MarshallYan** [comment]
> @MarshallYan , I am not sure if this modification is necessary.
> 
> I reverted the modification that you made to the `load_state_dict` method and outlined the 3 use cases in this [test](https://github.com/choderalab/modelforge/blob/b81639f015f7d912f66338ec801694a923d38119/modelforge/tests/test_models.py#L343). They are currently all passing. Could you please first add the use case that you are imagining and that is failing to this test and check that it is failing.

I went through the test and it did seem to pass. However, `trainer.model` may not be used as expected because:
 1. `trainer` should be a `ModelTrainer` object
 2. `ModelTrainer.model` is initialized by `ModelTrainer.setup_potential()`
 3. `ModelTrainer.setup_potential()` returns a `TrainingAdapter` object
 4. So, `trainer.model` is actually a `TrainingAdapter` object

Now, in the tests you linked to, since `trainer.model` is a `TrainingAdapter` object, and model2 is a `Potential` object, they actually called different `load_state_dict()` methods. This is super confusing to me but I agree that this shouldn't be fixed here.

So, in conclusion, I think we can simply close this PR and directly work on seperate two methods for creating models for training and inference.

**Comment by wiederm** [comment]
Merging this because it outlines the different use cases in the test.
